src/effects/hieroglyphs.c

```diff
@@ -8,7 +8,31 @@ INCLUDE_ASM(s32, "effects/hieroglyphs", hieroglyphs_main);
 void hieroglyphs_init(void) {
 }
 
-INCLUDE_ASM(s32, "effects/hieroglyphs", hieroglyphs_update);
+void hieroglyphs_update(EffectInstance* effect) {
+    HieroglyphsFXData* data = effect->data.hieroglyphs;
+    s32 temp_a2;
+
+    if (effect->flags & 16) {
+        effect->flags &= ~16;
+        data->unk_10 = 16;
+    }
+
+    if (data->unk_10 < 1000) {
+        data->unk_10--;
```

Comment:
we call these temps timeLeft (and the ones that count up every tick lifeTime)

Reply:
done